libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -370,6 +373,9 @@ bool AP_ICEngine::throttle_override(float &percentage)
         idle_percent > percentage)
     {
         percentage = idle_percent;
+        if (option_set(Options::THROTTLE_WHILE_DISARMED)) {
```

Comment:
Don't we need to check if disarmed?

Reply:
no, percentage==base_throttle when armed.

Reply:
It does at the moment. But not checking leaves a fairly massive bear trap for use to fail into in the future.